lib/rpmsg/rpmsg_virtio.c

```diff
@@ -376,8 +384,16 @@ static void *rpmsg_virtio_get_tx_payload_buffer(struct rpmsg_device *rdev,
 		metal_mutex_release(&rdev->lock);
 		if (rp_hdr || !tick_count)
 			break;
-		metal_sleep_usec(RPMSG_TICKS_PER_INTERVAL);
-		tick_count--;
+
+		/*
+		 * Try to use wait loop implemented in the virtio dispatcher and
+		 * use metal_sleep_usec() method by default.
+		 */
+		status = rpmsg_virtio_wait_notified(rvdev, rvdev->rvq);
+		if (status != RPMSG_SUCCESS) {
```

Comment:
If wait_notified is supported but failed, then user needs to know. In that case, we should print error message.

Also, fallback method should be used if RPMSG_EOPNOTSUPP (EXIO) is returned.

Reply:
> If wait_notified is supported but failed, then user needs to know. In that case, we should print error message.

What about letting the ops that implement the wait function print a more explicit error message instead of just printing an error code here? This would also have the advantage of not adding extra code in the library.

> 
> Also, fallback method should be used if RPMSG_EOPNOTSUPP (EXIO) is returned.
Good point , probably better to  call the metal_sleep_usec in case of error different from RPMSG_EOPNOTSUPP

Reply:
> What about letting the ops that implement the wait function print a more explicit error message instead of just printing an error code here? This would also have the advantage of not adding extra code in the library.
> 
> >

I am okay with this.